### PR TITLE
Add Math.Clamp

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add Clamp methods.
- Add new project for Unit Tests.
- Add Units tests for Math.Clamp.
- Update pipeline to run unit tests.
- Bump version to 1.4.3-preview.

## Motivation and Context
- Add Math.Clamp to nanoFramework Math API.
- Following pool to community. 
- Aligned with .NET dotnet/runtime#13994.

## How Has This Been Tested?
- Added Unit Tests for the Clamp methods.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by josesimoes** [comment]
### Build will fail because of Unit Tests. nanCLR is waiting for the update to the new Math assembly.

File: nanoFramework.System.Math/Math.cs [hunk 1]
```diff
@@ -333,5 +333,255 @@ public static int Sign(float value)
...
+        /// <returns>
+        /// <paramref name="value"/> if <paramref name="min"/> ≤ <paramref name="value"/> ≤ <paramref name="max"/>.
+        /// -or-
+        /// <paramref name="min"/> if <paramref name="value"/>&lt;<paramref name="min"/>.
+        /// -or-
+        /// max if <paramref name="max"/>&lt;<paramref name="value"/>.
+        /// </returns>
+        public static byte Clamp(byte value, byte min, byte max) 
+        {
+            if (value < min)
```
Comment (Ellerbach) [inline nanoFramework.System.Math/Math.cs:352]: to be faster, `if(value <= min)` you'll gain few instructions in this case

File: nanoFramework.System.Math/Math.cs [hunk 2]
```diff
@@ -333,5 +333,388 @@ public static int Sign(float value)
...
+        /// <returns>
+        /// <paramref name="value"/> if <paramref name="min"/> ≤ <paramref name="value"/> ≤ <paramref name="max"/>.
+        /// -or-
+        /// <paramref name="min"/> if <paramref name="value"/>&lt;<paramref name="min"/>.
+        /// -or-
+        /// max if <paramref name="max"/>&lt;<paramref name="value"/>.
+        /// </returns>
+        public static byte Clamp(byte value, byte min, byte max) 
+        {
+            if(max < min)
```
Comment (Ellerbach) [inline nanoFramework.System.Math/Math.cs:353]: I still think we can gain 1 check if max <= min
Let's take 1, 1, 2
at the first check if 1  < 1, you go for the second check 1 < 2 and will return 2
if 1 <= 1, then at the first check you'll return 1

**Review by Ellerbach — APPROVED** [review]
I love the double checks ;-)

**Pull request merged**